### PR TITLE
Support sub directory for artifact location

### PR DESCRIPTION
**Pull request by praveenkuttappan**
Java API review create step requires additional artifact sub directory param. This will help in using generic artifact path to look for config files.

### CODE REVIEW

File: eng/common/scripts/Create-APIReview.ps1
Comment (weshaggard) [inline eng/common/scripts/Create-APIReview.ps1:60]: Why create a new parameters? Why not adjust the java pipeline to just pass the correct path in as ArtifactPath?

Comment (praveenkuttappan) [reply eng/common/scripts/Create-APIReview.ps1:60]: I have made this change to pass root artifact directory only in artifact path since that's where all config json files are created before version is updated to alpha. Script needs to know config file location to parse them and find version. Other option is to create those json files in individual artifact locations based on group name and package name directories in the same way build is creating packages. But I thought It's probably simpler if we use additional param to generalize this for all languages and also If build step in any language clears target package directory it will be an issue too if config file is created in it(I am not sure if any lang does it though).

Comment (praveenkuttappan) [reply eng/common/scripts/Create-APIReview.ps1]: I have made the change as discussed to pass config path in new param.

**Comment by praveenkuttappan** [comment]
@weshaggard : Can you please have a look and approve this when you get a chance? I have made the change per our discussion to  use a separate param for config path

File: eng/common/scripts/Create-APIReview.ps1
Comment (weshaggard) [inline eng/common/scripts/Create-APIReview.ps1:89]:
```suggestion
    $ConfigFileDir = Join-Path -Path $ConfigFileDir "PackageInfo"
```

Lets move that there and if config is passed use it as passed.

File: eng/common/scripts/Create-APIReview.ps1
Comment (weshaggard) [inline eng/common/scripts/Create-APIReview.ps1:89]: I think you weren't understanding my other request. If ConfigFileDir is passed I'd expect it to be the exact directory where the config files exists and so we shouldn't append the PackageInfo path to those.

File: eng/common/scripts/Create-APIReview.ps1
Comment (weshaggard) [inline eng/common/scripts/Create-APIReview.ps1:95]: Why was this removed?

Comment (praveenkuttappan) [reply eng/common/scripts/Create-APIReview.ps1:95]: Package property config file is created only for new SDK type to avoid conflict between track1 and track2 packages within different group(java). This is per our discussion earlier to keep it simple. Pipeline step is marked as failed if package property file is missing for any package which will cause an issue when we enable this failure as hard error in the near future.

Comment (weshaggard) [reply eng/common/scripts/Create-APIReview.ps1:95]: OK I think I understand the logic now. I will say however this has the potential for track 2 libraries that for whatever reason that fail to publish a package info file from requiring approval.

**Review by weshaggard — APPROVED** [review]

File: eng/common/scripts/Create-APIReview.ps1
Comment (benbp) [inline eng/common/scripts/Create-APIReview.ps1:95]: nit: would `Join-Path -Path $ConfigFileDir "$PackageName.json"` read easier?

Comment (praveenkuttappan) [reply eng/common/scripts/Create-APIReview.ps1:95]: Thanks for this suggestion.

**Pull request merged**
